### PR TITLE
BMI323 and BMM350 implementation

### PR DESCRIPTION
**Pull request by jojos38**
- Working BMI323 at 400hz gyro and 200hz accel
- Working BMM350 with sensor fusion

It went ~100 hours of testing with no major issues so far, maybe some adjustments will be needed especially on the calibration side

### CODE REVIEW

File: src/configuration/CalibrationConfig.h
Comment (ButterscotchV) [inline src/configuration/CalibrationConfig.h:108]: Wouldn't adding a new entry in the middle break the saved configs for every sensor after?

File: src/sensors/bmi323sensor.cpp
Comment (ButterscotchV) [inline src/sensors/bmi323sensor.cpp:3]: I'm not sure this copyright is totally correct

File: src/sensors/bmi323sensor.h
Comment (ButterscotchV) [inline src/sensors/bmi323sensor.h:3]: I'm not sure this copyright is correct either

**Review by ButterscotchV — COMMENTED** [review]
Looks alright, I can't really review the IMU specific code but it looks alright from a skim over.

**Comment by jojos38** [comment]
I fixed the copyright

File: src/sensors/bmi323sensor.h
Comment (jojos38) [reply src/sensors/bmi323sensor.h]: I fixed the copyright

File: src/configuration/CalibrationConfig.h
Comment (jojos38) [reply src/configuration/CalibrationConfig.h]: I pushed a commit to put BMI323 at the end of the list

**Review by ButterscotchV — APPROVED** [review]

**Comment by ButterscotchV** [comment]
Looking to merge this soon, if anyone would like to review this or give feedback before it is merged, now is the time!

File: lib/bmi323/bmi323_jojos38.cpp
Comment (TheDevMinerTV) [inline lib/bmi323/bmi323_jojos38.cpp:1]: Please add copyright information here :)

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: lib/bmi323/bmi323_jojos38.cpp
Comment (Eirenliel) [reply lib/bmi323/bmi323_jojos38.cpp:1]: I understand wanting to immortalize your name, @jojos38, but please use copyright information for that and rename the files to bmi323.cpp/h and bmm350.cpp/h

**Review by Eirenliel — CHANGES_REQUESTED** [review]

File: lib/bmi323/bmi323_jojos38.cpp
Comment (jojos38) [reply lib/bmi323/bmi323_jojos38.cpp:1]: Actually the reason was that the original lib was already named bmi323.cpp and since I was working with both at the same time I renamed mine so that it doesn't conflict

**Comment by jojos38** [comment]
Hi, are the changes correct please?

File: src/sensors/bmi323sensor.cpp
Comment (TheDevMinerTV) [inline src/sensors/bmi323sensor.cpp:1]: Please run the VSCode formatter over this file and replace the remaining `Serial.print*()`s with `logger.trace()` calls (check the other sensors for initialization code).

File: src/sensors/bmi323sensor.h
Comment (TheDevMinerTV) [inline src/sensors/bmi323sensor.h:1]: Please run the VSCode formatter over this file.

File: src/sensors/bmi323sensor.h
Comment (TheDevMinerTV) [inline src/sensors/bmi323sensor.h:144]: Make these into `#define`s, `constexpr`s or `static`s, otherwise they get recreated with every instance of the sensor. It's not a lot of memory but still.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

**Comment by unlogisch04** [comment]
As discussions in discord,.. show that the BMI323 works bad and only about 5 ppl have sets, i'm not sure if it is a good idea to integrate this into main. Want to hear other opinions on it.

**Comment by jojos38** [comment]
> As discussions in discord,.. show that the BMI323 works bad and only about 5 ppl have sets, i'm not sure if it is a good idea to integrate this into main. Want to hear other opinions on it.

Replying a bit late sorry, while the BMI323 isn't the best sensor out there it's about the same as the BMI160 in term of tracking quality

**Comment by jojos38** [comment]
Closed because I implemented sfusion, will reopen a new PR

**Pull request closed without merging**
